<?xml version="1.0" ?>
<launch>
  
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="57600"/>
  </node>


  <rosparam param="base_width">0.34</rosparam>
  <rosparam param="ticks_meter">101</rosparam>



  <node name="lpid_velocity" pkg="rvbot_bringup" type="pid_velocity.py" output="screen">
      <remap from="wheel" to="lwheel"/>
      <remap from="motor_cmd" to="left_wheel_speed"/>
      <remap from="wheel_vtarget" to="lwheel_vtarget"/>
      <remap from="wheel_vel" to="lwheel_vel"/>   
      <rosparam param="Kp">400</rosparam>
      <rosparam param="Ki">100</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">-1023</rosparam>
      <rosparam param="out_max">1023</rosparam>
      <rosparam param="rate">30</rosparam>
      <rosparam param="timeout_ticks">4</rosparam>
      <rosparam param="rolling_pts">5</rosparam>
  </node>

  <node name="rpid_velocity" pkg="rvbot_bringup" type="pid_velocity.py" output="screen">
      <remap from="wheel" to="rwheel"/>
      <remap from="motor_cmd" to="right_wheel_speed"/>
      <remap from="wheel_vtarget" to="rwheel_vtarget"/>
      <remap from="wheel_vel" to="rwheel_vel"/>
      <rosparam param="Kp">400</rosparam>
      <rosparam param="Ki">100</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">-1023</rosparam>
      <rosparam param="out_max">1023</rosparam>
      <rosparam param="rate">30</rosparam>
      <rosparam param="timeout_ticks">4</rosparam>
      <rosparam param="rolling_pts">5</rosparam>
 </node>

  <node pkg="rvbot_bringup" type="twist_to_motors.py" name="twist_to_motors" output="screen"/>
  <node pkg="rvbot_bringup" type="diff_tf.py" name="diff_tf" output="screen"/> 
  <node pkg="rvbot_bringup" type="wheel_vel_pub.py" name="wheel_vel_pub" output="screen"/>

  <include file="$(find rvbot_bringup)/launch/irl_spawn.launch"/>
  <include file="$(find rvbot_bringup)/launch/ydlidar.launch"/>
  <!--<include file="$(find rvbot_navigation)/launch/gmapping.launch"/>-->


</launch>